frontend/src/components/activePage/TableWorkspaceLogic/TableWorkspaceLogic.tsx

```diff
@@ -36,179 +33,93 @@ export interface ITableWorkspaceLogicProps {
-      const templateList = Array.from(
-        new Set(
-          dataInstances?.instanceList?.instances?.map(
-            i =>
-              i?.spec?.templateCrownlabsPolitoItTemplateRef?.templateWrapper
-                ?.itPolitoCrownlabsV1alpha2Template?.spec?.templateName
-          )
-        )
-      );
-      const instancesMapped = dataInstances?.instanceList?.instances?.map(
-        (instance, index) => {
-          const { metadata, spec, status } = instance!;
-          const {
-            environmentList,
-            templateName,
-          } = spec?.templateCrownlabsPolitoItTemplateRef?.templateWrapper?.itPolitoCrownlabsV1alpha2Template?.spec!;
-          const {
-            firstName,
-            lastName,
-          } = spec?.tenantCrownlabsPolitoItTenantRef?.tenantWrapper?.itPolitoCrownlabsV1alpha1Tenant?.spec!;
-          return {
-            id: index,
-            name: metadata?.name,
-            gui: environmentList?.[0]?.guiEnabled,
-            persistent: environmentList?.[0]?.persistent,
-            idTemplate: spec?.templateCrownlabsPolitoItTemplateRef?.name!,
-            templatePrettyName: templateName,
-            ip: instance?.status?.ip,
-            status: instance?.status?.phase,
-            url: status?.url,
-            timeStamp: metadata?.creationTimestamp,
-            tenantId: spec?.tenantCrownlabsPolitoItTenantRef?.tenantId,
-            tenantNamespace: `tenant-${spec?.tenantCrownlabsPolitoItTenantRef?.tenantId}`,
-            tenantDisplayName: `${firstName}\n${lastName}`,
-            workspaceId: spec?.templateCrownlabsPolitoItTemplateRef?.namespace,
-          } as Instance;
-        }
-      );
-      const templates = templateList.map(t => {
-        const instancesTmp = instancesMapped?.filter(
-          ({ templatePrettyName: tpn }) => tpn === t
-        );
-        return {
-          id: instancesTmp![0].idTemplate,
-          name: t,
-          gui: instancesTmp![0]?.gui,
-          persistent: instancesTmp![0]?.persistent,
-          resources: { cpu: 0, memory: '', disk: '' },
-          instances: instancesTmp,
-          workspaceId: instancesTmp![0].workspaceId,
-        } as Template;
+          notifyStatus(
+            instance.status?.phase!,
+            instance,
+            updateType!,
+            tenantNamespace,
+            WorkspaceRole.manager
+          );
+
+          const newItem = { ...prev };
+          setDataInstances(newItem);
+          return prev;
+        },
       });
-      setTemplatesMapped(templates);
     }
-  }, [
-    loadingInstances,
-    dataInstances?.instanceList,
-    userId,
-    errorInstances,
-    workspaces,
-    dataInstances,
-    tenantNamespace,
-  ]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [loadingInstances, subscribeToMoreInstances, tenantNamespace, userId]);
+
+  const instancesMapped = dataInstances?.instanceList?.instances
+    ?.map((i, n) => getManagerInstances(i!, n))
```

Comment:
```suggestion
    ?.map(getManagerInstances)
```

Reply:
it seems doesn't like it

Reply:
you have to adapt getManagerInstances, so that it can accept a nullable i